### PR TITLE
Improve handling of stderr output

### PR DESCRIPTION
**Pull request by nj0yeh**
Improve handling of stderr output.

Write correct error message when the build fails.

### CODE REVIEW

**Comment by davidfowl** [comment]
2 things:
- Take a look at this PR that just went in https://github.com/davidfowl/Micronetes/commit/afe111b06c73c2360cad8055f367037be5ec710f
- Redo the commits to be based on master without all the merges.

**Comment by nj0yeh** [comment]
>     * Take a look at this PR that just went in [afe111b](https://github.com/davidfowl/Micronetes/commit/afe111b06c73c2360cad8055f367037be5ec710f)

I am aware of it and talked to the author. I think my solution is a little bit cleaner (and it solves the error of not displaying the build error messages).

>     * Redo the commits to be based on master without all the merges.

Done.

**Comment by davidfowl** [comment]
> I am aware of it and talked to the author. I think my solution is a little bit cleaner 

From an API POV sure but it doesn't change the experience, the output is interleaved into a single output stream. The previous solution does this by default so all process launches were capturing both stdout and error. This only turns it back on for process launches for the project executable.

> (and it solves the error of not displaying the build error messages).

Yep this PR has 2 different changes.

File: src/Micronetes.Hosting/Infrastructure/ProcessUtil.cs [hunk 1]
```diff
@@ -72,9 +73,9 @@ internal static class ProcessUtil
                 {
                     if (e.Data != null)
                     {
-                        if (outputDataReceived != null)
```
Comment (davidfowl) [inline src/Micronetes.Hosting/Infrastructure/ProcessUtil.cs:75]: I think the logic should put everything in output data unless error data is specified. By default there's no value is splitting the streams here.

File: src/Micronetes.Hosting/ProcessRunner.cs [hunk 2]
```diff
@@ -104,11 +104,12 @@ private Task LaunchService(Application application, Service service)
 
                 var buildResult = ProcessUtil.Run("dotnet", $"build \"{service.Status.ProjectFilePath}\" /nologo",
                                                  outputDataReceived: data => service.Logs.OnNext(data),
+                                                 errorDataReceived: data => service.Logs.OnNext(data),
```
Comment (davidfowl) [inline src/Micronetes.Hosting/ProcessRunner.cs]: Then we can also undo this.

File: src/Micronetes.Hosting/ProcessRunner.cs [hunk 3]
```diff
@@ -186,6 +187,7 @@ void RunApplication(IEnumerable<(int Port, int BindingPort, string Protocol)> po
                             environmentVariables: environment,
                             workingDirectory: workingDirectory,
                             outputDataReceived: data => service.Logs.OnNext($"[{replica}]: {data}"),
+                            errorDataReceived: data => service.Logs.OnNext($"[{replica}]: {data}"),
```
Comment (davidfowl) [inline src/Micronetes.Hosting/ProcessRunner.cs]: And this.

**Review by davidfowl — CHANGES_REQUESTED** [review]
Made a couple comments

**Review by davidfowl — APPROVED** [review]

**Pull request merged**